### PR TITLE
add a sample to count the frequency of callee functions of arcgis-python-api repo

### PR DESCRIPTION
**Pull request by CMPeng**
Tl;Dr

Not sure if this sample should be uploaded to the power_users folder??

### CODE REVIEW

File: samples/02_power_users_developers/count_the_usage_frequency_of_api_methods.ipynb [hunk 1]
```diff
@@ -0,0 +1,986 @@
+{
```
Comment (Yongyao) [inline samples/02_power_users_developers/count_the_usage_frequency_of_api_methods.ipynb]: <p>Perhaps you want to explain why we want to differentiate <code>result</code> and <code>result2</code>. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/514/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (Yongyao) [inline samples/02_power_users_developers/count_the_usage_frequency_of_api_methods.ipynb]: <p>Just a thought, does it make more sense to look at the popularity for each module like raster and feature module? For example, some might be interested to see which raster function is the most popular.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/514/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='59'/>

Comment (Yongyao) [inline samples/02_power_users_developers/count_the_usage_frequency_of_api_methods.ipynb]: <p>Looks like both Shannon and the learn lesson team like to differentiate our users into <code>GIS Professionals</code> and <code>Data Scientists</code>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/514/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='61'/>

**Comment by Yongyao** [comment]
@CMPeng Very cool analysis. 👍 Please see my comments through reviewNB

**Comment by CMPeng** [comment]
Thanks @Yongyao 
Your comments are addressed in the new commit. CC @AtmaMani

**Comment by CMPeng** [comment]
Closing this issue, and re-opening a new issue for internal archiving https://github.com/ArcGIS/geosaurus/pull/2583

**Pull request closed without merging**